### ISSUE TITLE
[SpecDecode] [V4] MTP for Deepseek V4

### ISSUE THREAD

**Issue by johanna-rock-tt** [tenstorrent/tt-metal#44074]
- Base off a cleaned up version of dynamic MTP from DS V3
- Spec decode manager: no changes for new model
- MTP module
  - Change base decoder layer in pipeline
  - MTP stage
    - change dims
    - h_norm with 3 non-zero dims
    - Change concat to addition of e_norm + h_norm
    - Implement additional ops for hc_head (rsqrt(mean(x^2)), linear, mult, sigmoid, addition, sum)
- Tests
  - Use diff reference i/o; same tests, just call different model MTP stage

```
e = Emb(input_ids)                             # [T, 7168]
e = where(position == 0, 0, e)                 # [T, 7168]
e = enorm(e)                                   # [T, 7168]
h = hnorm(previous_hidden_states, axis=-1)     # [T, 4, 7168]
z = h_proj(h) + unsqueeze(e_proj(e), axis=-2)  # [T, 4, 7168]
y = mtp_decoder_block(z)                       # decoder excluded
dense = hc_head(y, hc_head_fn, scale, base)    # [T, 7168]
logits = LMHead(shared_head_norm(dense))       # [T, 129280]

Where hc_head is:
y = y.flatten(2).float()
rsqrt = torch.rsqrt(y.square().mean(-1, keepdim=True) + self.norm_eps)
mixes = F.linear(y, hc_fn) * rsqrt
pre = torch.sigmoid(mixes * hc_scale + hc_base) + self.hc_eps
y = torch.sum(pre.unsqueeze(-1) * y.view(shape), dim=2)
```
